net/extras/genInventoryFile.py

```diff
@@ -37,33 +39,37 @@ def writeInventoryEntry(self, outFd, config):
             cfg_entry = "{}={}\n".format(config.lower(), self.configInfo[config])
             outFd.write(cfg_entry)
 
-    def writeInventory(self, outFd):
-        outFd.write("[" + Inventory.groupName + "]\n")
-        self.nodeCount = 1
+    def writeInventory(self, outFd, groupName, groupRole):
+        outFd.write("[" + groupName + "]\n")
         connInfo = SafeDict(self.configInfo['CONNECTION_INFO'])
 
         # Add host entries in the inventory file
         for node in connInfo:
+            role = connInfo[node].get('role', 'worker')
+            if role != groupRole:
+                continue
             var_line = "node{} ".format(self.nodeCount)
             outFd.write(var_line)
             var_line = "ansible_ssh_host={} ".format(node)
             outFd.write(var_line)
-            var_line = "contiv_network_mode={} ".format(self.networkMode)
-            outFd.write(var_line)
             var_line = "control_interface={} ".format(connInfo[node]['control'])
             outFd.write(var_line)
-            var_line = "netplugin_if={} ".format(connInfo[node]['data'])
-            outFd.write(var_line)
-            var_line = "fwd_mode={}\n".format(self.fwdMode)
+            var_line = "netplugin_if={} \n".format(connInfo[node]['data'])
             outFd.write(var_line)
 
             self.nodeCount += 1
 
-        self.nodeCount -= 1
+        outFd.write("\n")
+
+    def writeACI(self, outFd):
+        outFd.write("[" + "all:vars]\n")
+        var_line = "fwd_mode={}\n".format(self.fwdMode)
```

Comment:
I've changed it to writeGlobalVars to reflect that it is writing all the global vars, not just ACI.